# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/Parser.cpp
# Commit: c21f692a0693
# Full Hash: c21f692a069302eca39bdb1342bc6acb9fa14a3b
# Author: Ashley Hauck <khyperia@mozilla.com>
# Date: 2019-05-10 09:50:15
# Description:
#   Bug 1550628 - Correctly handle OOM. r=iain
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D30597
# ==============================================================================

diff -r e754d9ad197d -r c21f692a0693 js/src/frontend/Parser.cpp
--- a/js/src/frontend/Parser.cpp	Thu May 09 20:21:28 2019 +0000
+++ b/js/src/frontend/Parser.cpp	Fri May 10 01:08:35 2019 +0000
@@ -7433,7 +7433,7 @@
   }
 
   ListNodeType statementList = handler_.newStatementList(wholeInitializerPos);
-  if (!argsbody) {
+  if (!statementList) {
     return null();
   }
   handler_.addStatementToList(statementList, exprStatement);
